# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1420669610 18000
#      Wed Jan 07 17:26:50 2015 -0500
# Node ID 04d6d592c61423eeb21a10d158578904b150a659
# Parent  146f1daa805be207b7db49001d447c26be4c00f6
avoid valgrind warning: conditional move depends on uninitialized variable

* ignpoif: Explicitly initialize and save local array PP.

diff --git a/liboctave/cruft/ranlib/ignpoi.f b/liboctave/cruft/ranlib/ignpoi.f
--- a/liboctave/cruft/ranlib/ignpoi.f
+++ b/liboctave/cruft/ranlib/ignpoi.f
@@ -94,27 +94,28 @@ C     .. External Functions ..
       EXTERNAL ranf,sexpo,snorm
 C     ..
 C     .. Intrinsic Functions ..
       INTRINSIC abs,alog,exp,float,ifix,max0,min0,sign,sqrt
 C     ..
 C     JJV added this for case: mu unchanged
 C     .. Save statement ..
       SAVE s, d, l, ll, omega, c3, c2, c1, c0, c, m, p, q, p0,
-     +     a0, a1, a2, a3, a4, a5, a6, a7, fact, muprev, muold
+     +     a0, a1, a2, a3, a4, a5, a6, a7, fact, pp, muprev, muold
 C     ..
 C     JJV end addition - I am including vars in Data statements
 C     .. Data statements ..
 C     JJV changed initial values of MUPREV and MUOLD to -1.0E37
 C     JJV if no one calls IGNPOI with MU = -1.0E37 the first time,
 C     JJV the code shouldn't break
       DATA muprev,muold/-1.0E37,-1.0E37/
       DATA a0,a1,a2,a3,a4,a5,a6,a7/-.5,.3333333,-.2500068,.2000118,
      +     -.1661269,.1421878,-.1384794,.1250060/
       DATA fact/1.,1.,2.,6.,24.,120.,720.,5040.,40320.,362880./
+      DATA pp/35*0.0/
 C     ..
 C     .. Executable Statements ..
 
       IF (mu.EQ.muprev) GO TO 10
       IF (mu.LT.10.0) GO TO 120
 C
 C     C A S E  A. (RECALCULATION OF S,D,LL IF MU HAS CHANGED)
 C
